services/slam/builtin/builtin_test.go

```diff
@@ -717,7 +717,8 @@ func TestCartographerDataProcess(t *testing.T) {
 		time.Sleep(time.Millisecond * time.Duration(validDataRateMS*2))
 		cancelFunc()
 
-		latestLoggedEntry := obs.All()[len(obs.All())-1]
+		allObs := obs.All()
+		latestLoggedEntry := allObs[len(allObs)-1]
```

Comment:
[nit] Do you want to revert this change?

Reply:
The original code caused a failed flaky test - probably because between calling `obs.All()` twice, one more log was created, and thus it threw an index out of bounds error. This fixed the problem.